File: frontend/src/app/core/models/Project.ts
import { Material } from "./Material";
import { Users } from "./Users";

export enum FloorLevel {
    REZ_DE_CHAUSSEE = "REZ_DE_CHAUSSEE",
    PREMIER_ETAGE = "PREMIER_ETAGE",
    DEUXIEME_ETAGE = "DEUXIEME_ETAGE",
   
  }
  export enum Quality {
    STANDING = "STANDING",
    HAUT_STANDING = "HAUT_STANDING",
   
  }

export class Project {
    id: number;
    projectName: string;
    creationDate:Date;
    totalPrice:number;
    lanDimensions:string;
    floorLevel:FloorLevel;
    quality:Quality;
    nbBedroom:number;
    nbKitchen:number;
    nbLiving:number;
    nbBathroom:number;
    material:Material[];
    user:Users;

}

File: backend/src/main/java/tn/esprit/dari3d/Entities/Project.java
package tn.esprit.dari3d.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    Long id;
     String projectName;
     Date creationDate;
     Date modificationDate;
     Double totalPrice;
    Integer landDimensions;
  @Enumerated(EnumType.STRING)
  FloorLevel floorLevel;
  @Enumerated(EnumType.STRING)
  Quality quality;
  int nbBedroom;
  int nbKitchen;
  int nbLiving;
  int nbBathroom;
  int nbGarage;
  int terrace;



  @JsonIgnore

    @ManyToOne
     Users user;
  @JsonIgnore

   @ManyToMany(cascade = CascadeType.ALL)
     Set<Material> materials;

}
